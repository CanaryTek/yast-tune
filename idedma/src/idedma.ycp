/**
 * File:
 *   clients/idedma.ycp
 *
 * Package:
 *   Configuration of IDE DMA mode
 *
 * Summary:
 *   Main file
 *
 * Authors:
 *   Ladislav Slezak <lslezak@suse.cz>
 *
 * $Id$
 *
 * Main file for IDE DMA mode configuration. Uses all other files.
 *
 */

{
    /***
     * <h3>Configuration of the IDE DMA mode</h3>
     */

    textdomain "idedma";

    import "Mode";
    import "Idedma";
    include "idedma/ui.ycp";

    /* The main () */
    y2milestone ("Idedma module started");
    y2milestone ("----------------------------------------");

    /* main ui function */
    any ret = nil;

    if (Mode::test == true)
    {
	// in test mode set some data and display only main dialog
	Idedma::set_test_data();
	ret = IdedmaAutoSequence();
    }
    else
    {
	ret = IdedmaSequence ();
    }

    y2debug ("ret == %1", ret);

    /* Finish */
    y2milestone ("Idedma module finished");
    return ret;
}
