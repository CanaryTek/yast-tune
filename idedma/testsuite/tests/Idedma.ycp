// IDE DMA configuration module
//
// Idedma:: module testsuite
//
// $Id$
//

{
include "testsuite.ycp";
import "Idedma";


// disable displaying errors in Report module
Report::DisplayErrors(false, 0);

map READ = $[ "probe" : $[ "ide" : [ $["device" : "ABC-13245", "dev_name" : "/dev/hda"], $["device" : "XYZ-6789", "dev_name" : "/dev/hdc"] ] ,
    "scsi" : [ $["class_id":262, "dev_name":"/dev/sr0", "dev_name2":"/dev/hdb", "device":"TOSHIBA DVD-ROM SD-M1402", "driver":"ide-scsi", "sub_class_id":2 ]] ]];

map EXEC = $[ "target" : $["bash_output" : $["stdout" : " using_dma    =  1 (on)", "exit" : 0 ] , "bash" : 0 ] ];

DUMP("Read test");
TEST(``(Idedma::Read()), [ READ, $[], EXEC ], nil );
DUMP(Idedma::get_ide_devices());


READ = add(READ, "sysconfig", $["hardware" : $[ "DEVICES_FORCE_IDE_DMA" : ""] ]);
EXEC = $["target" : $["bash_output" : $["stdout" : "Error", "exit" : -1 ] ] ];

DUMP("Read test2");
TEST(``(Idedma::Read()), [ READ, $[], EXEC ], nil );
DUMP(Idedma::get_ide_devices());


READ = add(READ, "sysconfig", $["hardware" : $[ "DEVICES_FORCE_IDE_DMA" : "/dev/hda:on"] ]);
EXEC = $[ "target" : $["bash_output" : $["stdout" : " using_dma    =  1 (on)", "exit" : 0 ] ] ];

DUMP("Read test3");
TEST(``(Idedma::Read()), [ READ, $[], EXEC ], nil );
DUMP(Idedma::get_ide_devices());


READ = add(READ, "sysconfig", $["hardware" : $[ "DEVICES_FORCE_IDE_DMA" : "/dev/hda:off"] ]);

DUMP("Read test4");
TEST(``(Idedma::Read()), [ READ, $[], EXEC ], nil );
DUMP(Idedma::get_ide_devices());


map WRITE = $[];
READ = add(READ, "init", $[ "scripts" : $[ "exists" : true, "runlevel" : $["boot.idedma":$["start":["B"], "stop":[]]], "comment" : $["boot.idedma":$["defstart":["B"], "defstop":[], "description":"Enable/disable DMA mode on IDE devices.", "provides":["boot.idedma"], "reqstart":[], "reqstop":[]]] ] ]);
map EXEC2 = $[
    "target": $[
	"bash_output": $[ "exit": 0, "stdout": "", "stderr": "", ],
	],
    ];

Idedma::set_dma("/dev/hda", "on");
Idedma::set_dma("/dev/hdc", "off");
DUMP("Write test1");
DUMP(Idedma::get_ide_devices());
TEST(``(Idedma::Write()), [ READ, WRITE, [EXEC, EXEC, EXEC2] ], nil );

Idedma::set_dma("/dev/hda", "udma5");
Idedma::set_dma("/dev/hdc", "udma2");
DUMP("Write test2");
DUMP(Idedma::get_ide_devices());
TEST(``(Idedma::Write()), [ READ, WRITE, [EXEC, EXEC, EXEC2] ], nil );

Idedma::set_dma("/dev/hda", nil);
Idedma::set_dma("/dev/hdx", "udma2");
DUMP("Write test3");
DUMP(Idedma::get_ide_devices());
TEST(``(Idedma::Write()), [ READ, WRITE, [EXEC, EXEC, EXEC2] ], nil );

Idedma::set_dma("/dev/hda", "nochange");
Idedma::set_dma("/dev/hdx", "udma2");
DUMP("Write test4");
DUMP(Idedma::get_ide_devices());
TEST(``(Idedma::Write()), [ READ, WRITE, [EXEC, EXEC, EXEC2] ], nil );

TEST(``(Idedma::get_current_dma_mode(nil)), [$[], $[], $[]], nil);
TEST(``(Idedma::get_current_dma_mode("")), [$[], $[], $[]], nil);
TEST(``(Idedma::get_current_dma_mode("mdma2 udma0 udma1 udma2")), [$[], $[], $[]], nil);
TEST(``(Idedma::get_current_dma_mode("mdma2 udma0 udma1 *udma2")), [$[], $[], $[]], nil);
TEST(``(Idedma::get_current_dma_mode("mdma2 udma0 *udma1 *udma2")), [$[], $[], $[]], nil);
TEST(``(Idedma::get_current_dma_mode("mdma2   udma0   *udma1   *udma2")), [$[], $[], $[]], nil);

// no DMA - should return all modes which can be set in yast2
TEST(``(Idedma::get_supported_dma_modes(nil)), [$[], $[], $[]], nil);
TEST(``(Idedma::get_supported_dma_modes("")), [$[], $[], $[]], nil);

TEST(``(Idedma::get_supported_dma_modes("mdma2 udma0 udma1 udma2")), [$[], $[], $[]], nil);
TEST(``(Idedma::get_supported_dma_modes("mdma2 udma0 udma1 *udma2")), [$[], $[], $[]], nil);
TEST(``(Idedma::get_supported_dma_modes("mdma2 udma0 *udma1 *udma2")), [$[], $[], $[]], nil);
TEST(``(Idedma::get_supported_dma_modes("mdma2   udma0   *udma1   *udma2")), [$[], $[], $[]], nil);
}
