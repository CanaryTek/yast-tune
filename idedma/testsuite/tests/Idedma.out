Read	.target.yast2 "classnames.ycp" nil
Read	.probe.ide [$["dev_name":"/dev/hda", "device":"ABC-13245"], $["dev_name":"/dev/hdc", "device":"XYZ-6789"]]
Read	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_ON nil
Read	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_OFF nil
Execute	.target.bash_output "/sbin/hdparm -d /dev/hda" $["exit":0, "stdout":" using_dma    =  1 (on)"]
Execute	.target.bash_output "/sbin/hdparm -d /dev/hdc" $["exit":0, "stdout":" using_dma    =  1 (on)"]
Return	true
Dump	[$["current_dma":true, "dev_name":"/dev/hda", "device":"ABC-13245", "device_type":"Unknown device type", "dma_on":nil], $["current_dma":true, "dev_name":"/dev/hdc", "device":"XYZ-6789", "device_type":"Unknown device type", "dma_on":nil]]
Read	.target.yast2 "classnames.ycp" nil
Read	.probe.ide [$["dev_name":"/dev/hda", "device":"ABC-13245"], $["dev_name":"/dev/hdc", "device":"XYZ-6789"]]
Read	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_ON ""
Read	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_OFF "hdc"
Execute	.target.bash_output "/sbin/hdparm -d /dev/hda" $["exit":-1, "stdout":"Error"]
Execute	.target.bash_output "/sbin/hdparm -d /dev/hdc" $["exit":-1, "stdout":"Error"]
Return	true
Dump	[$["current_dma":nil, "dev_name":"/dev/hda", "device":"ABC-13245", "device_type":"Unknown device type", "dma_on":nil], $["current_dma":nil, "dev_name":"/dev/hdc", "device":"XYZ-6789", "device_type":"Unknown device type", "dma_on":false]]
Read	.target.yast2 "classnames.ycp" nil
Read	.probe.ide [$["dev_name":"/dev/hda", "device":"ABC-13245"], $["dev_name":"/dev/hdc", "device":"XYZ-6789"]]
Read	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_ON "hda"
Read	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_OFF ""
Execute	.target.bash_output "/sbin/hdparm -d /dev/hda" $["exit":0, "stdout":" using_dma    =  1 (on)"]
Execute	.target.bash_output "/sbin/hdparm -d /dev/hdc" $["exit":0, "stdout":" using_dma    =  1 (on)"]
Return	true
Dump	[$["current_dma":true, "dev_name":"/dev/hda", "device":"ABC-13245", "device_type":"Unknown device type", "dma_on":true], $["current_dma":true, "dev_name":"/dev/hdc", "device":"XYZ-6789", "device_type":"Unknown device type", "dma_on":nil]]
Read	.target.yast2 "classnames.ycp" nil
Read	.probe.ide [$["dev_name":"/dev/hda", "device":"ABC-13245"], $["dev_name":"/dev/hdc", "device":"XYZ-6789"]]
Read	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_ON ""
Read	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_OFF "hda"
Execute	.target.bash_output "/sbin/hdparm -d /dev/hda" $["exit":0, "stdout":" using_dma    =  1 (on)"]
Execute	.target.bash_output "/sbin/hdparm -d /dev/hdc" $["exit":0, "stdout":" using_dma    =  1 (on)"]
Return	true
Dump	[$["current_dma":true, "dev_name":"/dev/hda", "device":"ABC-13245", "device_type":"Unknown device type", "dma_on":false], $["current_dma":true, "dev_name":"/dev/hdc", "device":"XYZ-6789", "device_type":"Unknown device type", "dma_on":nil]]
Dump	[$["current_dma":true, "dev_name":"/dev/hda", "device":"ABC-13245", "device_type":"Unknown device type", "dma_on":true], $["current_dma":true, "dev_name":"/dev/hdc", "device":"XYZ-6789", "device_type":"Unknown device type", "dma_on":false]]
Execute	.target.bash "/sbin/hdparm -d 1 /dev/hda" 0
Execute	.target.bash_output "/sbin/hdparm -d /dev/hda" $["exit":0, "stdout":" using_dma    =  1 (on)"]
Execute	.target.bash "/sbin/hdparm -d 0 /dev/hdc" 0
Execute	.target.bash_output "/sbin/hdparm -d /dev/hdc" $["exit":0, "stdout":" using_dma    =  1 (on)"]
Log	Device /dev/hdc: Can not set DMA to value 0
Write	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_ON "hda" true
Write	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_OFF "hdc" true
Read	.init.scripts.exists "boot.idedma" true
Read	.init.scripts.exists "boot.idedma" true
Read	.init.scripts.runlevel "boot.idedma" nil
Read	.init.scripts.comment "boot.idedma" nil
Execute	.target.bash "/sbin/insserv -d boot.idedma" 0
Return	false
Dump	[$["current_dma":true, "dev_name":"/dev/hda", "device":"ABC-13245", "device_type":"Unknown device type", "dma_on":false], $["current_dma":true, "dev_name":"/dev/hdc", "device":"XYZ-6789", "device_type":"Unknown device type", "dma_on":false]]
Execute	.target.bash "/sbin/hdparm -d 0 /dev/hda" 0
Execute	.target.bash_output "/sbin/hdparm -d /dev/hda" $["exit":0, "stdout":" using_dma    =  1 (on)"]
Log	Device /dev/hda: Can not set DMA to value 0
Execute	.target.bash "/sbin/hdparm -d 0 /dev/hdc" 0
Execute	.target.bash_output "/sbin/hdparm -d /dev/hdc" $["exit":0, "stdout":" using_dma    =  1 (on)"]
Log	Device /dev/hdc: Can not set DMA to value 0
Write	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_ON "" true
Write	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_OFF "hda hdc" true
Read	.init.scripts.exists "boot.idedma" true
Read	.init.scripts.exists "boot.idedma" true
Read	.init.scripts.runlevel "boot.idedma" nil
Read	.init.scripts.comment "boot.idedma" nil
Execute	.target.bash "/sbin/insserv -d boot.idedma" 0
Return	false
Dump	[$["current_dma":true, "dev_name":"/dev/hda", "device":"ABC-13245", "device_type":"Unknown device type", "dma_on":nil], $["current_dma":true, "dev_name":"/dev/hdc", "device":"XYZ-6789", "device_type":"Unknown device type", "dma_on":false]]
Execute	.target.bash "/sbin/hdparm -d 0 /dev/hdc" 0
Execute	.target.bash_output "/sbin/hdparm -d /dev/hdc" $["exit":0, "stdout":" using_dma    =  1 (on)"]
Log	Device /dev/hdc: Can not set DMA to value 0
Write	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_ON "" true
Write	.sysconfig.hardware.DEVICES_FORCE_IDE_DMA_OFF "hdc" true
Read	.init.scripts.exists "boot.idedma" true
Read	.init.scripts.exists "boot.idedma" true
Read	.init.scripts.runlevel "boot.idedma" nil
Read	.init.scripts.comment "boot.idedma" nil
Execute	.target.bash "/sbin/insserv -d boot.idedma" 0
Return	false
